### PR TITLE
azure-cli-extensions.azure-iot: init at 0.25.0

### PR DESCRIPTION
**Pull request by Mikutut**
This PR adds azure-iot extension of Azure CLI, which is necessary for performing any operations on Azure IoT infrastructure.

Link to extension's repository: https://github.com/Azure/azure-iot-cli-extension

## Things done

- Extended Azure CLI package's `extensions-manual` list with `azure-iot` extension
- Added new package: [`azure-iot-device`](https://pypi.org/project/azure-iot-device/) at 0.24.0 version, which is an official Azure IoT Device SDK for Python and a dependency of `azure-iot` Azure CLI extension
- Added new package: [`requests-unixsocket2`](https://pypi.org/project/requests-unixsocket2/), which is a dependency of aforementioned `azure-iot-device` package

---

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mikutut** [comment]
Result of `nixpkgs-review pr 346752` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>azure-cli-extensions.azure-iot</li>
    <li>azure-cli-extensions.azure-iot.dist</li>
    <li>python311Packages.azure-iot-device</li>
    <li>python311Packages.azure-iot-device.dist</li>
    <li>python311Packages.requests-unixsocket2</li>
    <li>python311Packages.requests-unixsocket2.dist</li>
    <li>python312Packages.azure-iot-device</li>
    <li>python312Packages.azure-iot-device.dist</li>
    <li>python312Packages.requests-unixsocket2</li>
    <li>python312Packages.requests-unixsocket2.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/azure-iot-device/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    deprecation
+    paho-mqtt
+    requests
+    requests-unixsocket2
+    janus
+    pysocks
+    typing-extensions
+  ];
+
+  doCheck = false;
```
Comment (katexochen) [inline pkgs/development/python-modules/azure-iot-device/default.nix:44]: Please leave a comment why we cannot run checks.

File: pkgs/by-name/az/azure-cli/extensions-manual.nix [hunk 2]
```diff
@@ -84,6 +84,32 @@
     meta.maintainers = with lib.maintainers; [ katexochen ];
   };
 
+  azure-iot = mkAzExtension rec {
```
Comment (katexochen) [inline pkgs/by-name/az/azure-cli/extensions-manual.nix:87]: Please try to keep alphabetical order.

File: pkgs/development/python-modules/azure-iot-device/default.nix (hunk 1 above)
Comment (Mikutut) [reply pkgs/development/python-modules/azure-iot-device/default.nix:45]: Fixed

File: pkgs/by-name/az/azure-cli/extensions-manual.nix (hunk 2 above)
Comment (Mikutut) [reply pkgs/by-name/az/azure-cli/extensions-manual.nix]: Fixed

**Review by katexochen — APPROVED** [review]

**Pull request merged**